### PR TITLE
Use namespace for permissions

### PR DESCRIPTION
**Pull request by rudigiesler**
Have the API use a namespace for its own permissions, and have it configurable through the settings file.

### CODE REVIEW

**Comment by rudigiesler** [comment]
@justinvdm Ready for review.

File: authapi/permissions.py [hunk 1]
```diff
@@ -237,10 +241,14 @@ def has_object_permission(self, request, view, obj):
...
-        permissions = find_permission(permissions, permission_type, object_id)
+        permissions = find_permission(
+            permissions, permission_type, object_id,
+            settings.PERMISSION_NAMESPACE)
         return permissions.exists()
 
-    def check_permissions(self, user, ptype, object_id):
+    def check_permissions(self, user, ptype, object_id, namespace):
+        if namespace != settings.PERMISSION_NAMESPACE:
+            return True
```
Comment (justinvdm) [inline authapi/permissions.py]: Do we need a test for this?

File: authapi/permissions.py [hunk 2]
```diff
@@ -19,7 +20,9 @@ def __init__(self, permission_type):
     def has_permission(self, permission, request, view):
         user = request.user
         permissions = get_user_permissions(user)
-        permissions = find_permission(permissions, self.permission_type)
+        permissions = find_permission(
+            permissions, self.permission_type,
+            namespace=settings.PERMISSION_NAMESPACE)
```
Comment (justinvdm) [inline authapi/permissions.py]: Is there a simple way we could test that `AllowPermission` and `AllowObjectPermission` ignore permissions for a different namespace?

**Comment by justinvdm** [comment]
Left two small comments, looks good :)

File: authapi/permissions.py (hunk 1 above)
Comment (rudigiesler) [inline authapi/permissions.py]: I think that the `test_permission_add_permission_read_access` does test this. It tests that a user without any permissions can add a permission with namespace `foo`. Do you mean something more explicit?

File: authapi/permissions.py (hunk 2 above)
Comment (rudigiesler) [inline authapi/permissions.py]: We could just test the object directly? Although that would require constructing a request and view...

File: authapi/permissions.py (hunk 1 above)
Comment (justinvdm) [inline authapi/permissions.py]: Ah, missed that test, thanks.

File: authapi/permissions.py (hunk 2 above)
Comment (justinvdm) [inline authapi/permissions.py]: If I'm looking correctly, it looks like these are covered by the things that use them, nvm the comment.

**Comment by justinvdm** [comment]
:+1:

File: authapi/permissions.py (hunk 2 above)
Comment (rudigiesler) [inline authapi/permissions.py]: Yes, for at least one of the views (probably multiple), we have tests that test generic `foo` namespaces as well as the `__auth__` namespaces.

**Pull request merged**